src/StreamJsonRpc.Shared/Reflection/MethodNameAttribute.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using Microsoft;
+
+namespace StreamJsonRpc
+{
+    /// <summary>
+    /// Attribute which allows a method to have an alternate name.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = false)]
+    public class MethodNameAttribute : Attribute
```

Comment:
_MethodNameAttribute_ is kind of generic. How about _JsonRpcMethodAttribute_? 
Later we can add more properties to it; as it wouldn't be limited to just the method name.